lib/starlark_compiler/ast.rb

```diff
@@ -101,13 +103,30 @@ def initialize(str)
       end
     end
 
+    class Variable < Node
+      attr_reader :var
+      def initialize(var)
+        raise "Only string type is allowed as a variable, got #{var.class}" unless var.is_a?(::String)
+        @var = var
+      end
+    end
+
     class Array < Node
       attr_reader :elements
       def initialize(elements)
         @elements = elements.map(&method(:node))
       end
     end
 
+    class Assignment < Node
+      attr_reader :name, :var
+      def initialize(name, var)
+        @name = name
+        raise "Unsupported type on rhs for assignment: #{var.class}" if [FunctionCall, MethodCall, Assignment].include?(var.class)
```

Comment:
this is fine for now, but eventually I should refactor things so expressions and statements are represented separately here, and then this could just ensure the rhs is an expression

Reply:
also, why would function or method calls be invalid rhs of the assignment operator? They're expressions, and all expressions should be bindable as an rvalue